### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by DaveNatalieTripArc** [microsoft/playwright-dotnet#2127]
**Context:**
- Playwright Version: 1.21.0
- Operating System: Windows
- .NET version: NET 6
- Browser: Chromium

**Describe the bug**

I am unable to run the install script.  After adding the Microsoft.Playwright package to my project, I try to run the install script `pwsh bin\Debug\net6.0\playwright.ps1 install`, but I get the following error.

```
MethodInvocationException: {MY PROJECT FOLDER}\bin\Debug\net6.0\playwright.ps1:4
Line |
   4 |  [Reflection.Assembly]::LoadFile($PlaywrightFileName) | Out-Null
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "LoadFile" with "1" argument(s): "Could not load file or assembly
     | '{MY PROJECT FOLDER}\bin\Debug\net6.0\Microsoft.Playwright.dll'. The
     | system cannot find the file specified."

InvalidOperation: {MY PROJECT FOLDER}\bin\Debug\net6.0\playwright.ps1:5
Line |
   5 |  exit [Microsoft.Playwright.Program]::Main($args)
     |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Unable to find type [Microsoft.Playwright.Program].
```

Microsoft.Playwright.dll does not exist anywhere in my project's bin folder.

**Comment by fmc91** [comment]
I am having the same issue on .NET 5. The error message I receive is identical.
- Playwright Version: 1.21.0
- Operating System: Windows
- .NET version: .NET 5
- Browser: Chromium

**Comment by mxschmitt** [comment]
@fmc91 @DaveNatalieTripArc which PowerShell version are you using? (`pwsh --version`)

You can try upgrading PowerShell to the latest version, if I remember correctly there were some issues on specific versions.

There are multiple ways of upgrading it, the easiest is imo installing it as a global dotnet tool `dotnet tool install --global PowerShell`: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2#winget

**Comment by DaveNatalieTripArc** [comment]
I am using Powershell version: 7,2,3

I am not sure that this is a PowerShell issue. After adding the nuget package, `Microsoft.Playwright.dll` is not in the bin/Debug/net6.0 directory at all.

**Comment by mxschmitt** [comment]
Are you able to provide a repro for us which we can run locally?

Yes its very likely to be not a PowerShell issue. More likely something with the build setup or the way how the dependency get installed.

**Comment by Ruud2000** [comment]
@DaveNatalieTripArc what is your project type?

I can reproduce this issue when I install the playwright nuget in a **library** project.  
Assume a library called `Tester`. When I build this project, then the `bin\debug\netx` folder includes:

- `.playwright` subfolder
-  `playwright.ps1`
- `Tester.deps.json`
- `Tester.dll`
- `Tester.pdb`

Note, it does not include `Microsoft.Playwright.dll`.

When I install playwright nuget in a **console app** project and build, then the `bin\debug\netx` folder includes the following (project name `ConsoleApp`:

- `.playwright` subfolder
- `ConsoleApp.deps.json`
- `ConsoleApp.dll`
- `ConsoleApp.exe`
- `ConsoleApp.pdb`
- `ConsoleApp.runtimeconfig.json`
- `Microsoft.Bcl.AsyncInterfaces.dll`
- `Microsoft.Playwright.dll`
-  `playwright.ps1`
- `System.Text.Json.dll`

**Comment by campersau** [comment]
@Ruud2000 that is the "new" behavior of .NET Core / .NET as it no longer copies all the dependency DLLs by default for library projects as it did for .NET Framework, or more generally where `<HasRuntimeOutput>` is `false`.
See https://github.com/dotnet/sdk/blob/ee0504f6b4b95ff76a505cd236020a86c1f1fc63/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L23-L30
https://github.com/dotnet/sdk/blob/ad1ca90ad286288970a4ae1ec77e85848fef5a24/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L312-L331

So if you need the DLL in the output directory of a library project, set the property  `<HasRuntimeOutput>true</HasRuntimeOutput>` in the project file or pass the property as argument when building via CLI `/p:HasRuntimeOutput=true`.

**Comment by DaveNatalieTripArc** [comment]
@Ruud2000, you are correct, I am installing playwright in a library project. I was not aware of the `<HasRuntieOutput>` flag that @campersau mentioned. 

Thanks for providing a workaround.

**Comment by mxschmitt** [comment]
Closing as per above with the workaround.

**Comment by IvanRastamanov** [comment]
@campersau , @DaveNatalieTripArc , @mxschmitt 
Hello guys, can you please help with the place where this property (HasRuntieOutput) is?
Cannot find corresponding paths in my file system.

**Comment by campersau** [comment]
> @campersau , @DaveNatalieTripArc , @mxschmitt 
> Hello guys, can you please help with the place where this property (HasRuntieOutput) is?
> Cannot find corresponding paths in my file system.

It's a msbuild property which you can set in a project file inside of a `PropertyGroup`, it isn't there by default.
You can read more about this in general here: https://learn.microsoft.com/visualstudio/msbuild/msbuild

**Comment by IvanRastamanov** [comment]
> > @campersau , @DaveNatalieTripArc , @mxschmitt
> > Hello guys, can you please help with the place where this property (HasRuntieOutput) is?
> > Cannot find corresponding paths in my file system.
> 
> It's a msbuild property which you can set in a project file inside of a `PropertyGroup`, it isn't there by default. You can read more about this in general here: https://learn.microsoft.com/visualstudio/msbuild/msbuild

Thank you, looks like problem was in a bit different part.
